src/lib.rs

```diff
@@ -36,11 +36,22 @@ impl Image {
 
             let luminance = 0.30 * r + 0.59 * g + 0.11 * b;
 
-            if luminance >= 0.0 {
-                *pixel = Rgb([0, 0, luminance as u8]);
+            if luminance < 0.0 {
+                let mut val = -luminance as u8;
+                if invert {
+                    val = 255 - val;
+                    *pixel = Rgb([255, val, val]);
```

Comment:
To appear red or blue (against the white bg), we leave 255 in that channel and reduce the others.